linera-explorer/src/lib.rs

```diff
@@ -0,0 +1,405 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module provides web files to run a block explorer from linera service node.
+
+mod graphql;
+mod js_utils;
+
+use graphql_client::reqwest::post_graphql;
+use linera_base::{crypto::CryptoHash, identifiers::ChainId};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use serde_wasm_bindgen::from_value;
+use std::str::FromStr;
+use url::Url;
+use wasm_bindgen::prelude::*;
+
+use graphql::{
+    applications::ApplicationsApplications, block::BlockBlock, blocks::BlocksBlocks, Applications,
```

Comment:
I would also like better names, but it is generated by the deriver, the best I can do is rename the query name by `BlockQuery` and it will produce a `BlockQueryBlock` type.
Also if we rename the query by `CertificateValueQuery` it would give `CertificateValueQueryBlock`, but the problem remains for the other queries.